finality-aleph/src/testing/proposal.rs

```diff
@@ -0,0 +1,270 @@
+use crate::{
+    data_io::{
+        get_proposal_status, AlephProposal, AuxFinalizationChainInfoProvider,
+        CachedChainInfoProvider, ChainInfoCacheConfig,
+        PendingProposalStatus::*,
+        ProposalStatus::{self, *},
+        UnvalidatedAlephProposal, MAX_DATA_BRANCH_LEN,
+    },
+    testing::client_chain_builder::ChainBuilder,
+    SessionBoundaries, SessionId, SessionPeriod,
+};
+
+pub use sp_core::hash::H256;
+use sp_runtime::traits::Block as BlockT;
+use std::sync::Arc;
+use substrate_test_runtime_client::{
+    runtime::{Block, Header},
+    DefaultTestClientBuilderExt, TestClient, TestClientBuilder, TestClientBuilderExt,
+};
+
+fn proposal_from_headers(headers: Vec<Header>) -> AlephProposal<Block> {
+    let num = headers.last().unwrap().number;
+    let hashes = headers.into_iter().map(|header| header.hash()).collect();
+    let unvalidated = UnvalidatedAlephProposal::new(hashes, num);
+    let session_boundaries = SessionBoundaries::new(SessionId(0), SessionPeriod(10000));
```

Comment:
Hmm, I'm not sure what do you mean. Since `AlephProposal` does not have a public constructor a must use a trick to construct it -- `10000` is just a "large number" here.

Reply:
I meant that you put kinda arbitrary setting in a function that is not expected to define such things. I would just add some constant above this function, like `TEST_SESSION_BOUNDARIES` or something like this and use it in the current function

Reply:
Ok, I created `const DUMMY_SESSION_LEN: u32 = 1_000_000;` -- the point though is that the session bounds are not important in this test, so it's a dummy constant that needs to be big enough only.

Reply:
Also, I cannot really make a `SessionBoundaries` const -- rust does not allow me to :P